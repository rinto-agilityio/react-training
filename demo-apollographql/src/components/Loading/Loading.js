import React from 'react'

import { LoadingStyle } from './Loading.style'

const Loading = () => (
  <LoadingStyle />
)

export default Loading
