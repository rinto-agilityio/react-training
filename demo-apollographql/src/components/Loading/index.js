import styled from 'styled-components'

export default styled.div`
  border: 4px solid #f3f3f3;
  border-top: 4px solid #555;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  animation: spin 1s linear infinite;
`
