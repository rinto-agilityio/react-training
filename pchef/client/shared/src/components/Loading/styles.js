// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS, COLORS } from '../../themes'

const styles = StyleSheet.create({
  // Wrapper
  wrapper: {
    position: 'fixed',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    justifyContent: 'center',
    zIndex: METRICS.modalZindex,
    backgroundColor: COLORS.lightBlack,
    ...METRICS.flexCenter,
  },
})

export default styles
