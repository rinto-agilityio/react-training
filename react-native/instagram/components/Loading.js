import React from "react";
import { Text } from "react-native";

class Loading extends React.Component {
  render() {
    return <Text>Loading screen...</Text>;
  }
}

export default Loading;
