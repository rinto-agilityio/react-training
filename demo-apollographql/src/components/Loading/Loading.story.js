import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Loading from './index'
import { GlobalStyle } from '../../theme/globalStyle' // For animation

storiesOf('Components', module)
  .add('Loading', () => (
    <>
      <GlobalStyle />
      <Loading />
    </>
  ))
