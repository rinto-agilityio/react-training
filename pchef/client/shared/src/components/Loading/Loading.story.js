// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Loading from '.'

storiesOf('Loading', module)
  .add('small', () => (
    <Loading size="small" />
  ))
  .add('large', () => (
    <Loading size="large" />
  ))
