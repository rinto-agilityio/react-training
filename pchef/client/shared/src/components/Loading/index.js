// Libs
import React from 'react'
import { View, ActivityIndicator } from 'react-native'

// Themes
import { COLORS } from '../../themes'

// Styles
import styles from './styles'

type Props = {
  size?: string,
}

const Loading = ({
  size = 'small',
}: Props) => (
  <View style={styles.wrapper}>
    <ActivityIndicator size={size === 'medium' ? 'small' : size} color={COLORS.baseGray} />
  </View>
)

Loading.defaultProps = {
  size: 'small',
}

export default Loading
