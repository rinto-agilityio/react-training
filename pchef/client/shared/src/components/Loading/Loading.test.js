// Components
import Loading from '.'

describe('Components', () => {
  describe('Loading', () => {
    it('should render Loading component', () => {
      const renderComponent = renderer.create(<Loading />)
      expect(renderComponent).toMatchSnapshot()
    })
  })
})
