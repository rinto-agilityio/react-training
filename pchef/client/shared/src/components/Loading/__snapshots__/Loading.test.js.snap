// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-drfeu3 r-bottom-1p0dtai r-justifyContent-1777fci r-left-1d2f490 r-position-1xcajam r-right-zchlnj r-top-ipm5af r-zIndex-1sg8ghl"
>
  <div
    aria-valuemax="1"
    aria-valuemin="0"
    className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
    role="progressbar"
  >
    <div
      className="css-view-1dbjc4n r-animationDuration-17bb2tj r-animationIterationCount-1muvv40 r-animationKeyframes-127358a r-animationTimingFunction-1ldzwu0"
      style={
        Object {
          "height": "40px",
          "width": "40px",
        }
      }
    >
      <svg
        height="100%"
        viewBox="0 0 32 32"
        width="100%"
      >
        <circle
          cx="16"
          cy="16"
          fill="none"
          r="14"
          strokeWidth="4"
          style={
            Object {
              "opacity": 0.2,
              "stroke": "#c5c5c5",
            }
          }
        />
        <circle
          cx="16"
          cy="16"
          fill="none"
          r="14"
          strokeWidth="4"
          style={
            Object {
              "stroke": "#c5c5c5",
              "strokeDasharray": 80,
              "strokeDashoffset": 60,
            }
          }
        />
      </svg>
    </div>
  </div>
</div>
`;
